src/Provider/AbstractProvider.php

```diff
@@ -707,7 +707,7 @@ protected function parseResponse(ResponseInterface $response)
      */
     protected function prepareAccessTokenResponse(array $result)
     {
-        if ($this->getAccessTokenResourceOwnerId()) {
+        if ($this->getAccessTokenResourceOwnerId() !== null) {
```

Comment:
The specific Scrutinizer message is:

> The expression `$this->getAccessTokenResourceOwnerId()` of type `string|null` is loosely compared to true; this is ambiguous if the string can be empty. You might want to explicitly use `!== null` instead.

I'm not sure how Scrutinizer thinks the function returns `string|null`, when the return type annotation for it clearly shows just `string`. We probably just want to ignore this Scrutinizer issue, or figure out a way to annotate the return type hint to show that it can be (and might be) an empty string.

Clearly, if it's an empty string, we don't want this check to pass, so `!== null` won't work for us.